### ISSUE TITLE
Chart is not working anymore since airflow released version 2.0.2

### ISSUE THREAD

**Issue by griseau** [airflow-helm/charts#165]
**What is the bug?**

Even if in the values.yaml file I have `tag: 2.0.1-python3.8`, airflow 2.0.2 is getting deployed on one of the pod. When deploying the scheduler is not running. If I try to access the UI, I have the following : 
![image](https://user-images.githubusercontent.com/48803465/115362645-084e2b00-a1c2-11eb-8a9e-4e8a0ce6b160.png)

Somehow, airflow 2.0.2 is getting installed somewhere causing issue with database upgrade.

**What are your Helm values?**
```yaml
airflow.image:
    repository: apache/airflow
    tag: 2.0.1-python3.8
    pullPolicy: IfNotPresent
    pullSecret: ""
    uid: 50000
    gid: 50000
```

**What is your Kubernetes Version?**:

```console
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.14", GitCommit:"5de7fd1f9555368a86eb0f8f664dc58055c17269", GitTreeState:"clean", BuildDate:"2021-01-18T09:31:01Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**What is your Helm version?**:

```console
version.BuildInfo{Version:"v3.5.2", GitCommit:"167aac70832d3a384f65f9745335e9fb40169dc2", GitTreeState:"dirty", GoVersion:"go1.16rc1"}
```

**Comment by thesuperzapper** [comment]
@griseau Clearly there is an issue with your values, you must be passing a tag other than `2.0.1-python3.8`, it's literally not possible that  the chart as in this repo is deploying `2.0.2`. (With the possibly exception of if airflow themselves screwed up the DockerHub images)

**Comment by griseau** [comment]
@thesuperzapper when running into a container `airflow version` I have 2.0.1 so the version is the correct one, only in the UI it's showing 2.0.2, bit disappointing. I tried to deploy the chart on a new cluster and I'm facing the same issue.

**Comment by griseau** [comment]
@thesuperzapper looks like airflow mix version between 2.0.1 and 2.0.2 : 
![image](https://user-images.githubusercontent.com/48803465/115366821-ec4c8880-a1c5-11eb-9d99-f70742b44d43.png)

But when running airflow version inside the container : 
![image](https://user-images.githubusercontent.com/48803465/115366963-09815700-a1c6-11eb-94c2-6757d8505de3.png)

**Comment by griseau** [comment]
They just released 2.0.2-python3.8 on dockerhub, so I tried, but have the following error on the init install-pip-packages : 
`ERROR: Can not combine '--user' and '--target'`

**Comment by ZeWaren** [comment]
I'm in the exact same situation this morning. Let me know if I can help.

**Comment by griseau** [comment]
@ZeWaren, as a workaround you can use 2.0.2-python3.8 image without extraPipPackages, it worked on my side. 

@thesuperzapper we have 2 different issues here : 
- the chart is not working anymore with 2.0.1-python3.8 (first issue I pointed out)
- extraPipPackages doesn't work with airflow 2.0.2. Do you want me to create another issue for that ?

**Comment by kohonen** [comment]
> @thesuperzapper looks like airflow mixed version between 2.0.1 and 2.0.2 :
> ![image](https://user-images.githubusercontent.com/48803465/115366821-ec4c8880-a1c5-11eb-9d99-f70742b44d43.png)
> 
> But when running airflow version inside the container :
> ![image](https://user-images.githubusercontent.com/48803465/115366963-09815700-a1c6-11eb-94c2-6757d8505de3.png)
> 
> But I checked docker hub, image 2.0.1-python3.8 was pushed 2 months ago ... So something I'm not understanding here

it seems that the new helm chart somehow upgrades the airflow version to 2.0.2, most probably via pip

**Comment by thesuperzapper** [comment]
Just to keep everyone updated, the issue was caused by the existence of new airflow version when installing packages like `"apache-airflow-providers-databricks==1.0.1"`, as pip seems to try and install the newer versions of airflow.

**Comment by ZeWaren** [comment]
> 
> 
> @ZeWaren, as a workaround you can use 2.0.2-python3.8 image without extraPipPackages, it worked on my side.
> 

I confirms 2.0.2 works without `extraPipPackages`. However, the thing is I need those packages for my DAGs. :slightly_smiling_face:

**Comment by griseau** [comment]
> Just to keep everyone updated, the issue was caused by the existence of new airflow version when installing packages like `"apache-airflow-providers-databricks==1.0.1"`, as pip seems to try and install the newer versions of airflow.

Thanks @thesuperzapper, then the use of constraint files as proposed in #143 would fix it.

**Comment by griseau** [comment]
Let me know if I can help on that

**Comment by griseau** [comment]
> > @ZeWaren, as a workaround you can use 2.0.2-python3.8 image without extraPipPackages, it worked on my side.
> 
> I confirms 2.0.2 works without `extraPipPackages`. However, the thing is I need those packages for my DAGs. 🙂

In the meantime, you can use 2.0.1 and explicitly add "apache-airflow==2.0.1" as an extraPipPackages. It should work :)

**Comment by thesuperzapper** [comment]
I have made two seperate issues which track the two issues discussed here, lets move discussion over to them:
* https://github.com/airflow-helm/charts/issues/169
* https://github.com/airflow-helm/charts/issues/168

NOTE: I am going to go to sleep, its 1am in Australia, but please discuss and work on a solution if you want.

**Comment by thesuperzapper** [comment]
Hi all, this should be resolved after version `8.0.9` of the chart.
